File: backend/src/main/java/com/example/doctor/appointment/entity/Schedule.java
package com.example.doctor.appointment.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

@Entity
@Data
@Table(
        name = "schedules",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"doctor_id", "date"})
        }
)
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;
    @OneToMany(mappedBy = "schedule",cascade = CascadeType.ALL)
    private List<TimeSlot> timeSlots;
    private int maxPatients;

    @OneToMany(mappedBy = "schedule",fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private List<Appointment> appointments;

}

File: frontend/src/types/Schedule.ts
import type { Doctor } from "./Doctor";

export interface TimeSlot {
  id: number
  time:string
  available: boolean;
  schedule: Schedule;
  bookedPatients: number
}

export interface Schedule {
  id: number;
  doctor: Doctor;
  date: string;
  startTime:string
  endTime:string
  maxPatients: number
  timeSlots: TimeSlot[];
}